### PR TITLE
task4598 detail post

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4598

### CODE REVIEW

File: app/Repositories/Eloquents/AnswerRepository.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        return $this->model()->all();
+    }
+
+    public function getAnswerFromPost($id)
+    {
+        return $this->model()->with('user')->where('post_id', $id)->get();
+    }
+
+    public function getRepliesFromAnswers($answers)
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/AnswerRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/AnswerRepository.php</br>**Line**: 38</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Repositories/Eloquents/AnswerRepository.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        $answersReplies = array();
+
+        foreach ($answers as $key => $ans) {
+            $answersReplies[$key] = $ans->replies()->with('user')->get();
+        }
+
+        return $answersReplies;
+    }
+
+    public function getVotesFromAnswers($answers)
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/AnswerRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.ShortArraySyntax</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/AnswerRepository.php</br>**Line**: 49</br>"array()" function used. Use short array syntax, "[]", instead.

File: app/Repositories/Eloquents/PostRepository.php [hunk 3]
```diff
@@ -158,4 +159,41 @@ public function search($query, $input, $limit, $isPaginate = false)
...
+        return $post->replies()->with('user')->get();
+    }
+
+    // get replies with posts
+    public function findVoteFromPost($post)
+    {
+        return $post->votes()->where(['voteable_id' => $post->id, 'user_id' => \Auth::id()])->first();
+    }
+
+    public function getRelatedPost($listTag, $post_id)
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 185</br>Variable "post_id" is not in valid camel caps format

File: app/Repositories/Eloquents/PostRepository.php [hunk 4]
```diff
@@ -158,4 +159,41 @@ public function search($query, $input, $limit, $isPaginate = false)
...
+    {
+        $relatedPosts = collect([]);
+        foreach ($listTag->pluck('id') as $key => $tag) {
+            $relatedPosts = $relatedPosts->merge(Tag::with('posts')->find($tag)->posts);
+        }
+
+        $relatedPosts = $relatedPosts->unique('id');
+        $relatedPosts = $relatedPosts->filter(function($item) use ($post_id) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterFunction</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 193</br>Expected 1 space after FUNCTION keyword; 0 found

Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 193</br>Variable "post_id" is not in valid camel caps format

File: app/Repositories/Eloquents/PostRepository.php [hunk 5]
```diff
@@ -158,4 +159,41 @@ public function search($query, $input, $limit, $isPaginate = false)
...
+            return $item->id != $post_id;
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 194</br>Variable "post_id" is not in valid camel caps format

File: app/Http/Controllers/PostController.php [hunk 6]
```diff
@@ -21,42 +21,70 @@ class PostController extends Controller
      * Author: Lampham
      */
     protected $postRepository;
+    protected $answerRepository;
+    protected $tagRepository;
+    protected $userRepository;
+    protected $report;
 
-    public function __construct(PostRepositoryInterface $postRepository)
+    public function __construct(PostRepositoryInterface $postRepository,AnswerRepositoryInterface $answerRepository,
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.FirstParamSpacing</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 29</br>The first parameter of a multi-line function declaration must be on the line after the opening bracket

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.OneParamPerLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 29</br>Multi-line function declarations must define one parameter per line

File: app/Http/Controllers/PostController.php [hunk 7]
```diff
@@ -21,42 +21,70 @@ class PostController extends Controller
...
+        TagRepositoryInterface $tagRepository, UserRepositoryInterface $userRepository, Report $report)
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.OneParamPerLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 30</br>Multi-line function declarations must define one parameter per line

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.OneParamPerLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 30</br>Multi-line function declarations must define one parameter per line

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 30</br>The closing parenthesis of a multi-line function declaration must be on a new line

File: app/Http/Controllers/PostController.php [hunk 8]
```diff
@@ -21,42 +21,70 @@ class PostController extends Controller
...
     {
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.NewlineBeforeOpenBrace</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 31</br>The closing parenthesis and the opening brace of a multi-line function declaration must be on the same line

File: app/Http/Controllers/PostController.php [hunk 9]
```diff
@@ -21,42 +21,70 @@ class PostController extends Controller
...
+        $postReplies = $this->postRepository->getRepliesFromPost($post);
+        $postVote = $this->postRepository->findVoteFromPost($post);
+        //get answer from post id
+        $answers = $this->answerRepository->getAnswerFromPost($id);
+        $answersReplies = $this->answerRepository->getRepliesFromAnswers($answers);
+        $answersVotes = $this->answerRepository->getVotesFromAnswers($answers);
+        $relatedPost = $this->postRepository->getRelatedPost($post->tags, $id);
+
+        if (!empty($post)) {
+            return view('home.post.detail', compact('post', 'postReplies', 'postVote',
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 84</br>Opening parenthesis of a multi-line function call must be the last content on the line

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.MultipleArguments</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 84</br>Only one argument is allowed per line in a multi-line function call

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.MultipleArguments</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 84</br>Only one argument is allowed per line in a multi-line function call

File: app/Http/Controllers/PostController.php [hunk 10]
```diff
@@ -21,42 +21,70 @@ class PostController extends Controller
...
+                'answers', 'answersReplies', 'answersVotes', 'relatedPost'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 85</br>Multi-line function call not indented correctly; expected 12 spaces but found 16

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.MultipleArguments</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 85</br>Only one argument is allowed per line in a multi-line function call

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.MultipleArguments</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 85</br>Only one argument is allowed per line in a multi-line function call

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.MultipleArguments</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 85</br>Only one argument is allowed per line in a multi-line function call

Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 85</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Repositories/Eloquents/PostRepository.php [hunk 11]
```diff
@@ -158,4 +159,41 @@ public function search($query, $input, $limit, $isPaginate = false)
...
+
+    public function getRelatedPost($listTag, $postId)
+    {
+        $relatedPosts = collect([]);
+        foreach ($listTag->pluck('id') as $key => $tag) {
+            $relatedPosts = $relatedPosts->merge(Tag::with('posts')->find($tag)->posts);
+        }
+
+        $relatedPosts = $relatedPosts->unique('id');
+        $relatedPosts = $relatedPosts->filter(function($item) use ($postId) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.SpaceAfterFunction</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 193</br>Expected 1 space after FUNCTION keyword; 0 found

File: app/Http/Controllers/PostController.php [hunk 12]
```diff
@@ -21,42 +21,75 @@ class PostController extends Controller
...
+        $postVote = $this->postRepository->findVoteFromPost($post);
+        //get answer from post id
+        $answers = $this->answerRepository->getAnswerFromPost($id);
+        $answersReplies = $this->answerRepository->getRepliesFromAnswers($answers);
+        $answersVotes = $this->answerRepository->getVotesFromAnswers($answers);
+        $relatedPost = $this->postRepository->getRelatedPost($post->tags, $id);
+
+        if (!empty($post)) {
+            return view(
+            'home.post.detail',
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 89</br>Multi-line function call not indented correctly; expected 16 spaces but found 12

File: app/Http/Controllers/PostController.php [hunk 13]
```diff
@@ -21,42 +21,75 @@ class PostController extends Controller
...
+            compact('post', 'postReplies', 'postVote', 'answers', 'answersReplies', 'answersVotes', 'relatedPost'));
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 90</br>Closing parenthesis of a multi-line function call must be on a line by itself

**Comment by laam55** [comment]
ready

**Pull request merged**